File: 42578.cpp
// 위장

#include <string>
#include <vector>
#include <map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    map<string, int> strM;

    for (int i = 0; i < clothes.size(); i++) {
        strM[clothes[i][1]]++;
    }
    int answer = 1;
    map<string, int>::iterator hash;
    for (hash = strM.begin(); hash != strM.end(); hash++) {
        answer *= hash -> second + 1;
    }

    return answer - 1;
}

File: 42578.py
# 위장
# 2021-09-20

def solution(clothes):
    cat, clist = [], []
    answer = 1
    for cloth in clothes:
        if cloth[1] in cat:
            tmp = cat.index(cloth[1])
            clist[tmp] += 1
        else:
            cat.append(cloth[1])
            clist.append(1)

    for c in clist:
        answer *= (c + 1)

    return answer - 1

print(solution([["yellowhat", "headgear"], ["bluesunglasses", "eyewear"], ["green_turban", "headgear"]]))